styles/style_ptr.h

```diff
@@ -133,14 +140,20 @@ class ChargingStyle : public Style<T> {
 // Get a pointer to class.
 template<class STYLE>
 StyleAllocator StylePtr() {
+  using _TopBaseT  = typename style_base_check_detail::TopBase<STYLE>::type;
```

Comment:
Now that Layers<> can't become magically opaque anymore, we can remove this, right?
Edit:.. see below first